pyface/fields/i_editable_field.py

```diff
@@ -0,0 +1,62 @@
+# (C) Copyright 2005-2023 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" The editable field interface. """
+
+
+from traits.api import HasTraits
+
+from .i_field import IField
+
+
+class IEditableField(IField):
+    """ The editable field interface.
+
+    A editable field is a widget that displays a user-editable value.
+    """
+
+
+class MEditableField(HasTraits):
+    """The editable field mix-in.
+
+    Classes which use this mixin should implemnt _observe_control_value to
```

Comment:
```suggestion
    Classes which use this mixin should implement _observe_control_value to
```

Reply:
It looks like classes using this mixin should also implement `_get_control_value` (either directly, or indirectly - e.g., through also inheriting from `MField` or `Field`). Should that also be documented here?

Would it make sense to have `MEditableField` inherit from `MField`?

Reply:
`Field` requires `_get_control_value` (it's used for testing - we need a way in tests to see what value is set), so it should

Regarding the inheritance, I've been trying to keep the mixins from inherting from each other.  IIRC a few years ago I ran into an issue where Trait declarations were wrong when using inheritance in these mixins because trait definitions aren't evaluated following `mro`. (This is the relevant issue, I think https://github.com/enthought/traits/issues/401)